### ISSUE TITLE
ts_tensorflow.ipynb - Implementation # 1

### ISSUE THREAD

**Issue by shivankurkapoor** [YitongCU/Duel-staged-Attention-for-NYC-Weather-prediction#2]
Hi Yitong, thanks for the code. Can you please answer my following question:
In the following two methods, why do you select states[1]/s[1] from the LSTMStateTuple (c,h) where c is the hidden state and h is the output ? Shouldn't it be states[0]/s[0]. I referred to this paper [https://arxiv.org/pdf/1704.02971.pdf](url). 
   
    def en_RNN(self,input_x):
        with tf.variable_scope('fw_lstm') as scope:
            cell = self.encode_cell  ## this step don't create variable 
            out, states = tf.nn.dynamic_rnn(cell, input_x, self.seq_len, dtype = tf.float32) ## this step does create variables
            
        tmp = tf.tile(states[1],[1,self.time_step])
        tmp = tf.reshape(tmp,[self.batch_size, self.time_step, self.n_hidden])
        
        concat = tf.concat([out,tmp],axis = 2)
        return concat ## shape shoule be (b,T,2*n_hidden)
    
    def de_RNN(self,h):
        
        with tf.variable_scope('dec_lstm') as scope:
            cell = self.decode_cell
            d,s = tf.nn.dynamic_rnn(cell, h, self.seq_len, dtype = tf.float32)
        tmp = tf.tile(s[1],[1,self.time_step])
        tmp = tf.reshape(tmp,[self.batch_size, self.time_step, self.d_hidden])
        concat = tf.concat([d,tmp],axis = 2)
        return d,concat

**Comment by NicolaDonelli** [comment]
I agree. I think that the way it is implemented in the notebook basically concatenates two times the outputs of the cells.

**Comment by wangyazhao001** [comment]
> Hi Yitong, thanks for the code. Can you please answer my following question:
> In the following two methods, why do you select states[1]/s[1] from the LSTMStateTuple (c,h) where c is the hidden state and h is the output ? Shouldn't it be states[0]/s[0]. I referred to this paper [https://arxiv.org/pdf/1704.02971.pdf](url).
> 
> ```
> def en_RNN(self,input_x):
>     with tf.variable_scope('fw_lstm') as scope:
>         cell = self.encode_cell  ## this step don't create variable 
>         out, states = tf.nn.dynamic_rnn(cell, input_x, self.seq_len, dtype = tf.float32) ## this step does create variables
>         
>     tmp = tf.tile(states[1],[1,self.time_step])
>     tmp = tf.reshape(tmp,[self.batch_size, self.time_step, self.n_hidden])
>     
>     concat = tf.concat([out,tmp],axis = 2)
>     return concat ## shape shoule be (b,T,2*n_hidden)
> 
> def de_RNN(self,h):
>     
>     with tf.variable_scope('dec_lstm') as scope:
>         cell = self.decode_cell
>         d,s = tf.nn.dynamic_rnn(cell, h, self.seq_len, dtype = tf.float32)
>     tmp = tf.tile(s[1],[1,self.time_step])
>     tmp = tf.reshape(tmp,[self.batch_size, self.time_step, self.d_hidden])
>     concat = tf.concat([d,tmp],axis = 2)
>     return d,concat
> ```

Hi,brother.I also find the problem.How can you solve it ?
